odoo/orm/fields_relational.py

```diff
@@ -569,6 +621,59 @@ def _check_sudo_commands(self, comodel):
             return comodel.sudo(False).with_user(comodel.env.transaction.default_env.uid)
         return comodel
 
+    def condition_to_sql(self, model: BaseModel, alias: str, field_expr: str, operator: str, value, query: Query) -> SQL:
+        assert field_expr == self.name, "Supporting condition only to field"
+        model._check_field_access(self, 'read')
+        comodel = model.env[self.comodel_name]
+        if self.auto_join:
+            # bypass access rules for auto_join
+            comodel = comodel.sudo()
```

Comment:
It doesn't sound correct to me (or your code prevent subdomain somewhere):
If the value is a Domain containing other subdomain, these subdomain shouldn't be tranform in SQL/execute with sudo.
Only the current comodel restriction should be bypass.